### PR TITLE
Fixed default visibility.

### PR DESCRIPTION
**Pull request by akallabeth**
When nothing is declared, only export symbols defined
with WINPR_API or FREERDP_API defined.
Override this setting if BUILD_TESTING to allow tests
access to internal functions usually not exposed.

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot do you need a vacation or why do you not test?

File: server/shadow/Mac/mac_shadow.c [hunk 1]
```diff
@@ -660,7 +660,7 @@ macShadowSubsystem* mac_shadow_subsystem_new()
 	return subsystem;
 }
 
-int Mac_ShadowSubsystemEntry(RDP_SHADOW_ENTRY_POINTS* pEntryPoints)
+FREERDP_API int Mac_ShadowSubsystemEntry(RDP_SHADOW_ENTRY_POINTS* pEntryPoints)
```
Comment (realjiangms) [inline server/shadow/Mac/mac_shadow.c]: We might not want to export Win_ShadowSubsystemEntry/ X11_ShadowSubsystemEntry /Mac_ShadowSubsystemEntry. Can we still leave them internal currently?

**Comment by realjiangms** [comment]
+1.  We sometimes compile well at Linux but fail on Windows because of difference on symbol export between Windows and Linux.

**Comment by realjiangms** [comment]
There are still some redundant FREERDP_API in wf_interface.c:
wf_interface.c:215:FREERDP_API BOOL wfreerdp_server_is_running(wfServer* server)
wf_interface.c:232:FREERDP_API UINT32 wfreerdp_server_num_peers()
wf_interface.c:242:FREERDP_API UINT32 wfreerdp_server_get_peer_hostname(int pId, wchar_t * dstStr)
wf_interface.c:267:FREERDP_API BOOL wfreerdp_server_peer_is_local(int pId)
wf_interface.c:287:FREERDP_API BOOL wfreerdp_server_peer_is_connected(int pId)
wf_interface.c:308:FREERDP_API BOOL wfreerdp_server_peer_is_activated(int pId)
wf_interface.c:328:FREERDP_API BOOL wfreerdp_server_peer_is_authenticated(int pId)
wf_interface.c:348:FREERDP_API void wfreerdp_server_register_callback_event(cbCallback cb)

The FREERDP_API signature had already been defined in wf_interface.h.
Would you also fix them conveniently?

File: server/shadow/Mac/mac_shadow.c (hunk 1 above)
Comment (akallabeth) [inline server/shadow/Mac/mac_shadow.c]: @realjiangms No, it will not link any more.

Comment (realjiangms) [inline server/shadow/Mac/mac_shadow.c]: That's exactly why I have PR #3167 :)

But you are right, please keep it in your PR for now :)
If #3167 is merged first, you can remove them in this PR.
If this PR is merged first, I can remove them in my PR. 
Just make sure FreeRDP compiles well at anytime :)

**Comment by bmiklautz** [comment]
@akallabeth nice work!
+1

**Pull request merged**